File: spark-scala/spark-dataset/spark-dataset-typed/src/main/scala/com/mahfooz/spark/dataset/typed/transformation/filter/DatasetFilter.scala
package com.mahfooz.spark.dataset.typed.transformation.filter

import com.mahfooz.spark.dataset.model.Flight
import org.apache.spark.sql.SparkSession

object DatasetFilter {

  def main(args: Array[String]): Unit = {

    val sparkWarehouse =
      sys.env.getOrElse("SPARK_WAREHOUSE", "spark-warehouse").replace("\\", "/")

    val spark = SparkSession
      .builder()
      .master("local[*]")
      .appName("DatasetFilter")
      .config("spark.sql.warehouse.dir", sparkWarehouse)
      .getOrCreate()

    import spark.implicits._

    val dataPath =
      s"file:///${sys.env.getOrElse("DATA_HOME", "data").replace("\\", "/")}/FileData/Parquet/Flight/2010-summary.parquet"

    val flightsDF = spark.read.parquet(dataPath)

    val flights = flightsDF.as[Flight]

    flights.filter(flight => flight.count > 20).show(false)
  }

}

File: spark-java/spark-dataset/src/main/java/com/mahfooz/spark/dataset/filter/DatasetFilter.java
package com.mahfooz.spark.dataset.filter;

import com.mahfooz.spark.dataset.row.DatasetRow;
import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;
import org.apache.spark.sql.SparkSession;

public class DatasetFilter {

    public static void main(String[] args) {

        String fileName = args[0];

        SparkSession sparkSession = SparkSession
                .builder()
                .master("local")
                .appName(DatasetRow.class.getSimpleName())
                .getOrCreate();

        // Create an Dataset of Person objects from a text file
        Dataset<Row> dataset = sparkSession.read()
                .option("header", true)
                .csv(fileName);

        dataset.printSchema();

        dataset.filter("count == 20")
                .show();
    }
}
